openvdb/tools/FastSweeping.h

```diff
@@ -0,0 +1,1432 @@
+        auto hashComp2 = [&](uint32_t a, uint32_t b){return hash(coords[a]) < hash(coords[b]);};
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Alternative sort");
+#endif
+        tbb::parallel_sort(&tmp[0], &tmp[0] + mParent->voxelCount(), hashComp2);
+
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Sorting by sweep plane");
+#endif
+        auto hashComp = [&hash](const Coord &a, const Coord &b){return hash(a) < hash(b);};
+        tbb::parallel_sort(coords, coords + mParent->voxelCount(), hashComp);
+
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Computing number of sweep planes");
+#endif
+        util::PagedArray<size_t, 6> planes;
+        this->buildPlanes(hash, planes);
+
+        auto range = [&](size_t i) {
+          static const size_t min = 16, size = 100*std::thread::hardware_concurrency();
+          const size_t grainSize = std::max((planes[i] - planes[i-1])/size, min);
+          //const size_t grainSize = 1024;//2048;//512;//seems very sensitive to the grain size!
+          return tbb::blocked_range<size_t>(planes[i-1], planes[i], grainSize);
+        };
+
+#if 1
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Forward  sweep");
+#endif
+        for (size_t i = 1; i < planes.size(); ++i) tbb::parallel_for(range(i), *this);
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Backward sweeps");
+#endif
+        for (size_t i = planes.size()-1; i>0; --i) tbb::parallel_for(range(i), *this);
+#else
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.restart("Forward and backward sweeps");
+#endif
+        tbb::parallel_invoke([&](){for (size_t i = 1; i < planes.size(); ++i) tbb::parallel_for(range(i), *this);},
+                             [&](){for (size_t i = planes.size()-1; i>0; --i) tbb::parallel_for(range(i), *this);});
+#endif
+#ifdef BENCHMARK_FAST_SWEEPING
+        timer.stop();
+#endif
+    }
+    template<typename HashOp, typename ArrayType>
+    void buildPlanes(HashOp &hash, ArrayType &planes)
+    {
+        using BufferT = typename ArrayType::ValueBuffer;
+        using PoolT = tbb::enumerable_thread_specific<BufferT>;
+        BufferT exemplar(planes);//dummy used for initialization
+        PoolT pool(exemplar);//thread local storage pool of ValueBuffers
+        auto func = [&](const tbb::blocked_range<size_t>& r) {
+            Coord *p = mParent->mCoords.get();
+            typename PoolT::reference b = pool.local();
+            for (size_t i=r.begin(); i!=r.end(); ++i) if (hash(p[i-1])!=hash(p[i])) b.push_back(i);
+        };
+        planes.push_back(0);
+        tbb::parallel_for(tbb::blocked_range<size_t>(1, mParent->voxelCount(), 1<<12), func);// 4096
+        for (auto iter=pool.begin(); iter!=pool.end(); ++iter) iter->flush();
+        planes.push_back(mParent->voxelCount());
+        planes.sort();
+    }
+    /// @brief Locally solves @f$|\nabla \phi|^2 = 1 @f$ by means
+    /// of Godunov's upwind finite difference scheme
+#if 0
```

Comment:
Can you elaborate on this disabled block?

Reply:
an old block of code that will be removed (as replaced by the other fork of the #else